### DISCUSSION TITLE
ImportError: cannot import name dataclass_transform

### DISCUSSION THREAD

**Discussion by GodotIsWaitingToo** [gdsfactory/gdsfactory#1645] [General]
I have installed gdsfactory and tried to install the klayout-integration in terminal (I'm on MacOS Ventura) but it failed with the above error message. 

I then added the gdsfactory integration through the KLayout GUI (Tools --> Manage Packages). 

However, when in VSCode (having added the GDSFactory plugin and Python, pointing the path to the mambaforge Python path) I run an example from the tutorials, I get the error message again. Screenshot of the full error stack attached.

Any help getting up and running would be much appreciated!

Thanks!

<img width="1261" alt="Screenshot 2023-05-15 at 20 53 56" src="https://github.com/gdsfactory/gdsfactory/assets/26734990/27108e82-8f5d-4f79-b527-24bbc805c4fd">

**Comment by lucasgrjn** [comment]
Hi,

It looks like you issue would likely come from pydantic instead of gdsfactory.
From https://github.com/ray-project/ray/issues/31227, a  possible solution may be to update typing_extensions.
Can you try it ?

Regards,
Lucas.

**Comment by joamatab** [comment]
Hi,

what version of gdsfactory, pydantic and python are you using?

can you run this and copy the output here?

```python
import sys
import pydantic
from gdsfactory.config import print_version

print_version()
print(sys.version)
print(pydantic.__version__)
```

**Reply by GodotIsWaitingToo** [reply]
This is where I've got to, but see below for full update. Thank you joamatab!


<img width="822" alt="Screenshot 2023-05-17 at 11 22 29" src="https://github.com/gdsfactory/gdsfactory/assets/26734990/8b4f35c1-14d4-4aed-a061-c4d33a6bc67c">

**Comment by GodotIsWaitingToo** [comment]
Thank you both - I did the upgrade to typing_extensions as Lucas suggested, and I also reinstalled gdsfactory. I had a problem installing the klayout extension directly from terminal, so I ended up doing it in klayout using the GUI, which seems to have worked. Thank you for your help!
